Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -14,60 +14,72 @@ In the following section, we list some common root detection methods you'll enco
 
 Root detection can also be implemented through libraries such as [RootBeer](https://github.com/scottyab/rootbeer "RootBeer").
 
-#### SafetyNet
+#### Google Play Integrity
+  
+Google has launched the Google Play Integrity API to improve the security and integrity of apps and games on Android starting from Android 4.4 (level 19). The previous official API, [SafetyNet](https://developer.android.com/training/safetynet), did not cover all the security needs that Google wanted for the platform, so Play Integrity was developed with the basic functions of the previous API and integrated additional features. This change aims to protect users against dangerous and fraudulent interactions.
 
-SafetyNet is an Android API that provides a set of services and creates profiles of devices according to software and hardware information. This profile is then compared to a list of accepted device models that have passed Android compatibility testing. Google [recommends](https://developers.google.com/android/reference/com/google/android/gms/safetynet/SafetyNet "SafetyNet Documentation") using the feature as "an additional in-depth defense signal as part of an anti-abuse system".
+**Google Play Integrity provides the following protections:**
 
-How exactly SafetyNet works is not well documented and may change at any time. When you call this API, SafetyNet downloads a binary package containing the device validation code provided from Google, and the code is then dynamically executed via reflection. An [analysis by John Kozyrakis](https://koz.io/inside-safetynet/ "SafetyNet: Google's tamper detection for Android") showed that SafetyNet also attempts to detect whether the device is rooted, but exactly how that's determined is unclear.
+Verification of genuine Android device: verifies that the app is running on a legitimate Android device.  
+User license validation: Indicates whether the user installed or paid for the app or game via the Google Play Store.  
+Unmodified binary verification: determines if the app is interacting with the original binary recognized by Google Play.
 

```

Comment:
I think this could be reworded as:

Google Play Integrity offers the following safeguards:

- **Verification of genuine Android device**: It verifies that the application is running on a legitimate Android device.
- **User license validation**: It indicates whether the user installed or purchased the application or game through the Google Play Store
- **Unmodified binary verification**: It determines whether the application is interacting with the original binary recognized by Google Play.

Reply:
Fixed as requested